File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        
        for i in range(len(matrix)):
            if(target == matrix[i][0] or target == matrix[i][len(matrix[i])-1]):
                return True
            if target > matrix[i][0] and target < matrix[i][len(matrix[i])-1]:
                arr = matrix[i]
                left = 0
                right = len(matrix[i]) - 1

                while(left<=right):
                    mid = ((right-left) // 2) + left

                    if(arr[mid] == target):
                        return True
                    elif(arr[mid] > target):
                        right = mid - 1
                    elif(arr[mid] < target):
                        left = mid + 1
        return False
                

File: Data Structures & Algorithms/rotate-matrix/submission-0.js
class Solution {
    /**
     * @param {number[][]} matrix
     * @return {void}
     */
    rotate(matrix) {
        
        //make rows into cols
        for(let i = 0;i < matrix.length;i++) {

            for(let j = i + 1;j < matrix[i].length;j++) {
                //i = 0, j = 1
                let temp = matrix[j][i]
                matrix[j][i] = matrix[i][j];
                matrix[i][j] = temp 
            }
        }

        //reverse the cols by reversing the rows
        for(let i = 0;i < matrix.length;i++) {
            let left = 0
            let right = matrix[i].length - 1
            while(left<=right) {
                let temp = matrix[i][left]
                matrix[i][left] = matrix[i][right]
                matrix[i][right] = temp
                left++
                right--
            }
        }


        return matrix
        
    }
}
